src/app/main.rs

```diff
@@ -516,8 +536,7 @@ where
                     debug!("outbound dst={:?}", addr);
                     addr
                 }))
-                .layer(buffer::layer(max_in_flight))
-                .layer(pending::layer())
+                .buffer_pending(max_in_flight, DispatchDeadline::extract)
```

Comment:
TIOLI: this seems to be repeated enough that maybe there should be a `buffer_with_deadline` helper or something? on the other hand that might not really make the code any simpler.